services/order.py

```diff
@@ -0,0 +1,29 @@
+from django.contrib.auth import get_user_model
+
+from db.models import Order, Ticket, MovieSession
+from django.db import transaction
+
+
+@transaction.atomic
+def create_order(
+    tickets: list[dict], username: str, date: str | None = None
+) -> None:
+    user = get_user_model().objects.get(username=username)
+    order = Order.objects.create(user=user)
```

Comment:
Instead of checking if date is set and updating the order.created_at after creation, you can set created_at directly in the create() call